netlogo-gui/src/main/lab/gui/Supervisor.scala

```diff
@@ -74,21 +75,31 @@ class Supervisor(dialog: java.awt.Dialog,
       val fileName = org.nlogo.swing.FileDialog.showFiles(
         workspace.getFrame, "Exporting as spreadsheet", java.awt.FileDialog.SAVE,
         workspace.guessExportName(worker.protocol.name + "-spreadsheet.csv"))
-      addExporter(new SpreadsheetExporter(
+      try{
+	    addExporter(new SpreadsheetExporter(
         workspace.getModelFileName,
         workspace.world.getDimensions,
         worker.protocol,
-        new java.io.PrintWriter(new java.io.FileWriter(fileName))))
+		new java.io.PrintWriter(new java.io.FileWriter(fileName))))
+	  }
+	  catch{
+		case e: IOException => failure(e); return
+      }
     }
     if(options.table) {
       val fileName = org.nlogo.swing.FileDialog.showFiles(
         workspace.getFrame, "Exporting as table", java.awt.FileDialog.SAVE,
         workspace.guessExportName(worker.protocol.name + "-table.csv"))
-      addExporter(new TableExporter(
+      try{
+	  addExporter(new TableExporter(
         workspace.getModelFileName,
         workspace.world.getDimensions,
         worker.protocol,
         new java.io.PrintWriter(new java.io.FileWriter(fileName))))
+	  }
+	  catch{
+		case e: IOException => failure(e); return
+      }
```

Comment:
Similar comments here as above, make sure indentation uses spaces, make sure things are properly aligned, and either split the catch block up or move it to one line.